modules/OCPP201/conversions.cpp

```diff
@@ -1513,5 +1516,48 @@ to_ocpp_clear_display_message_response(const types::display_message::ClearDispla
     return result_response;
 }
 
+types::iso15118::EnergyTransferMode
+to_everest_allowed_energy_transfer_mode(const ocpp::v2::EnergyTransferModeEnum& allowed_energy_transfer_mode) {
+    switch (allowed_energy_transfer_mode) {
+    case ocpp::v2::EnergyTransferModeEnum::AC_BPT:
+        return types::iso15118::EnergyTransferMode::AC_BPT;
+    case ocpp::v2::EnergyTransferModeEnum::AC_BPT_DER:
+        return types::iso15118::EnergyTransferMode::AC_BPT_DER;
+    case ocpp::v2::EnergyTransferModeEnum::AC_DER:
+        return types::iso15118::EnergyTransferMode::AC_DER;
+    case ocpp::v2::EnergyTransferModeEnum::AC_single_phase:
+        return types::iso15118::EnergyTransferMode::AC_single_phase_core;
+    case ocpp::v2::EnergyTransferModeEnum::AC_three_phase:
+        return types::iso15118::EnergyTransferMode::AC_three_phase_core;
+    case ocpp::v2::EnergyTransferModeEnum::AC_two_phase:
+        return types::iso15118::EnergyTransferMode::AC_two_phase;
+    case ocpp::v2::EnergyTransferModeEnum::DC:
+        return types::iso15118::EnergyTransferMode::DC;
+    case ocpp::v2::EnergyTransferModeEnum::DC_BPT:
+        return types::iso15118::EnergyTransferMode::DC_BPT;
+    case ocpp::v2::EnergyTransferModeEnum::DC_ACDP:
+        return types::iso15118::EnergyTransferMode::DC_ACDP;
+    case ocpp::v2::EnergyTransferModeEnum::DC_ACDP_BPT:
+        return types::iso15118::EnergyTransferMode::DC_ACDP_BPT;
+    case ocpp::v2::EnergyTransferModeEnum::WPT:
+        return types::iso15118::EnergyTransferMode::WPT;
+    }
+    throw std::out_of_range("Could not convert EnergyTransferModeEnum");
+}
+
+std::optional<std::vector<types::iso15118::EnergyTransferMode>> to_everest_allowed_energy_transfer_modes(
+    const std::optional<std::vector<ocpp::v2::EnergyTransferModeEnum>>& allowed_energy_transfer_modes) {
```

Comment:
More in line with the other implementations in this file it would be to not pass an optional but the actual vector here and check for `has_value` in the function that calls it.